ext/standard/type.c

```diff
@@ -23,11 +23,18 @@ PHP_FUNCTION(gettype)
 {
 	zval *arg;
 	zend_string *type;
+	zend_bool resolve_object_names = 0;
 
-	ZEND_PARSE_PARAMETERS_START(1, 1)
+	ZEND_PARSE_PARAMETERS_START(1, 2)
 		Z_PARAM_ZVAL(arg)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_BOOL(resolve_object_names)
 	ZEND_PARSE_PARAMETERS_END();
 
+	if (resolve_object_names && Z_TYPE_P(arg) == IS_OBJECT) {
+		RETURN_STR_COPY(Z_OBJ_P(arg)->ce->name);
```

Comment:
One gotcha I see with this solution is the ambiguous result when a `resource` class instance is passed to `get_type()`. See https://3v4l.org/5aEd2 for example. I admit though that this is an edge case (and maybe `resource` should be reserved?).